src/sdk/resources/sites.ts

```diff
@@ -1,9 +1,20 @@
 import { IHttpClient } from '../http';
 import { EditableResource } from './resource';
 import { RemoteSite } from '../models';
+import { IDownloadFileProvider} from './resource';
+import { FileDownloader} from '../../util/file-downloader';
 
-export class SitesResource extends EditableResource<RemoteSite> {
+export class SitesResource extends EditableResource<RemoteSite> implements IDownloadFileProvider {
 	constructor(httpClient: IHttpClient) {
 		super(httpClient, 'site');
 	}
+
+	download(id: string, outputDirectory: string): any {
```

Comment:
I would remove this method and rely on callers to use the `FileDownloader` for themselves. The main reason is because if we are going to leverage the sdk pieces in the browser, file download would not be available in that case. So, I would like to keep the main file-download concerns/implementation out of the `sdk` directory.